tests/components/iqvia/test_config_flow.py

```diff
@@ -1,42 +1,48 @@
 """Define tests for the IQVIA config flow."""
+from unittest.mock import patch
+
+from pyiqvia.errors import InvalidZipError
+
 from homeassistant import data_entry_flow
 from homeassistant.components.iqvia import CONF_ZIP_CODE, DOMAIN
 from homeassistant.config_entries import SOURCE_USER
 from homeassistant.core import HomeAssistant
 
-
-async def test_duplicate_error(hass: HomeAssistant, config, config_entry) -> None:
-    """Test that errors are shown when duplicates are added."""
-    result = await hass.config_entries.flow.async_init(
-        DOMAIN, context={"source": SOURCE_USER}, data=config
-    )
-    assert result["type"] == data_entry_flow.FlowResultType.ABORT
-    assert result["reason"] == "already_configured"
+from .conftest import TEST_ZIP_CODE
 
 
-async def test_invalid_zip_code(hass: HomeAssistant) -> None:
-    """Test that an invalid ZIP code key throws an error."""
-    result = await hass.config_entries.flow.async_init(
-        DOMAIN, context={"source": SOURCE_USER}, data={CONF_ZIP_CODE: "bad"}
-    )
-    assert result["type"] == data_entry_flow.FlowResultType.FORM
-    assert result["errors"] == {CONF_ZIP_CODE: "invalid_zip_code"}
-
-
-async def test_show_form(hass: HomeAssistant) -> None:
-    """Test that the form is served with no input."""
+async def test_create_entry(hass: HomeAssistant, config, mock_pyiqvia) -> None:
+    """Test creating an entry."""
     result = await hass.config_entries.flow.async_init(
         DOMAIN, context={"source": SOURCE_USER}
     )
     assert result["type"] == data_entry_flow.FlowResultType.FORM
     assert result["step_id"] == "user"
 
+    # Test a bad ZIP code as input:
+    with patch(
+        "homeassistant.components.iqvia.config_flow.Client", side_effect=InvalidZipError
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input=config
+        )
+        assert result["type"] == data_entry_flow.FlowResultType.FORM
+        assert result["step_id"] == "user"
+        assert result["errors"] == {CONF_ZIP_CODE: "invalid_zip_code"}
+
+    # Test that we can recover from the error:
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input=config
+    )
+    assert result["type"] == data_entry_flow.FlowResultType.CREATE_ENTRY
```

Comment:
Config flow tests should patch `async_setup_entry`, preferably with a `mock_setup_entry` fixture in `conftest.py` and coresponding `pytestmark`
See #88484